### PR TITLE
Added component resource for building an S3 serverless static site

### PR DESCRIPTION
**Pull request by blarghmatey**
- Added component resource to build a serverless site with S3, ACM, and Cloudfront
- Updated s3 infrastructure module to build a serverless site to host the certificates

### CODE REVIEW

File: src/ol_infrastructure/components/aws/s3_cloudfront_site.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+
+class S3ServerlessSiteConfig(AWSBase):
+    """Configuration object for customizing a static site hosted with S3 and Cloudfront."""
+
+    site_name: Text
+    domains: List[Text]
+    bucket_name: Text
+    site_index: Text = 'index.html'
+    cloudfront_price_class: CloudfrontPriceClass = CloudfrontPriceClass.us_eu
```
Comment (shaidar) [inline src/ol_infrastructure/components/aws/s3_cloudfront_site.py:30]: Should that price class be hardcoded or is there a way to override it?

File: src/ol_infrastructure/components/aws/s3_cloudfront_site.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            validation_method='DNS',
+            subject_alternative_names=site_config.domains[1:]
+        )
+
+        s3_origin_id = f'{site_config.site_name}-s3-origin'
+
+        self.cloudfront_distribution = cloudfront.Distribution(
+            f'{site_config.site_name}-cloudfront-distribution',
+            aliases=site_config.domains,
+            comment='Cloudfront distribution for serving legacy xPRO certificates',
```
Comment (shaidar) [inline src/ol_infrastructure/components/aws/s3_cloudfront_site.py:80]: Shouldn't that be coming from `site_config.comment` instead?

File: src/ol_infrastructure/components/aws/s3_cloudfront_site.py (hunk 1 above)
Comment (blarghmatey) [reply src/ol_infrastructure/components/aws/s3_cloudfront_site.py:30]: It's not hardcoded, that's just a default. This acts as a data class which gets instantiated for the specific instances.

File: src/ol_infrastructure/components/aws/s3_cloudfront_site.py (hunk 2 above)
Comment (blarghmatey) [reply src/ol_infrastructure/components/aws/s3_cloudfront_site.py:80]: Yes it should

**Comment by shaidar** [comment]
👍

**Pull request merged**